packages/core/src/data-access/documentVersions/getDocumentByPath.ts

```diff
@@ -0,0 +1,34 @@
+import { database } from '$core/client'
+import { NotFoundError, Result } from '$core/lib'
+
+import { getDocumentsAtCommit } from './getDocumentsAtCommit'
+
+export async function getDocumentByPath(
+  {
+    projectId,
+    commitUuid,
+    path,
+  }: {
+    projectId: number
+    commitUuid: string
+    path: string
+  },
+  db = database,
+) {
+  try {
+    const result = await getDocumentsAtCommit({ projectId, commitUuid }, db)
+    const documents = result.unwrap()
+    const document = documents.find((doc) => doc.path === path)
+    if (!document) {
+      return Result.error(
+        new NotFoundError(
+          `No document with path ${path} at commit ${commitUuid}`,
+        ),
+      )
+    }
+
+    return Result.ok(document)
+  } catch (err) {
+    return Result.error(err as Error)
+  }
+}
```

Comment:
for now the api is allowed to fetch documents from staged commits, we can reevaluate this later on if we want to restrict it to main branch stuff.